File: in_Python/912A_Halloumi_boxes.py
t = int(input())
for i in range(t):
    num,max = map(int,input().split())
    arr = list(map(int,input().split()))
    
    status = 1
    for i in range(1,num):
        if arr[i] < arr[i-1]:
            status = 0
            break
    if status == 1:
        print("YES")
    elif max >= 2:
        print("YES")
    else:
        print("NO")

File: in_C++/912A_Halloumi_boxes.cpp
#include <bits/stdc++.h>
#include<algorithm>
using namespace std;

int main() {
    int t;
    cin >> t;
    while(t--) {
        int num,max;
        cin >> num >> max;

        int arr[num];
        for (int i = 0;i < num;i++) {
            cin >> arr[i];
        }
        int status = 0;
        if(is_sorted(arr,arr+num)) {
            status = 1;
        }
        if (status == 1) {
            cout <<  "YES" << endl;
        }
        else if (max >= 2) {
            cout << "YES" << endl;
        }
        else {
            cout << "NO" << endl;
        }
    }
    return 0;
}